--- a/luci-app-turboacc/Makefile
+++ b/luci-app-turboacc/Makefile
@@ -37,11 +37,11 @@ config PACKAGE_$(PKG_NAME)_INCLUDE_OFFLOADING
 config PACKAGE_$(PKG_NAME)_INCLUDE_SHORTCUT_FE_DRV
 	bool "Include Shortcut-FE for ECM"
 	depends on PACKAGE_$(PKG_NAME)_INCLUDE_SHORTCUT_FE=n
-	default y if (TARGET_ipq806x||TARGET_ipq807x)
+	default n
 
 config PACKAGE_$(PKG_NAME)_INCLUDE_SHORTCUT_FE
 	bool "Include Shortcut-FE"
-	default y if !(TARGET_ipq806x||TARGET_ipq807x||TARGET_ramips||i386||x86_64)
+	default y if !(TARGET_ramips||i386||x86_64)
 
 config PACKAGE_$(PKG_NAME)_INCLUDE_BBR_CCA
 	bool "Include BBR CCA"
